DUMBFILE_SYSTEM.md

```diff
@@ -1,30 +1,35 @@
 Specification of DUMBFILE_SYSTEM
 ================================
 
+DUMB is designed filesystem-agnostic, even though the C standard library
+already defines an abstraction over files on a disk. This is useful because
+Allegro 4 and 5 define their own abstractions.
+
+To register your own filesystem abstraction with DUMB, you must create an
+instance of struct `DUMBFILE_SYSTEM`, fill in your own function pointers
+according to the specification below, and call `register_dumbfile_system` on
+your instance.
+
 The header `dumb.h` defines `DUMBFILE_SYSTEM` as a struct of function pointers:
 
 ```
 typedef struct DUMBFILE_SYSTEM
 {
     void *(*open)(const char *filename);
-    int (*skip)(void *f, long n);
+    int (*skip)(void *f, dumb_off_t n);
```

Comment:
Maybe typedef the callback types, and then use those in the struct ? It would make things look cleaner imo.

Reply:
A matter of taste I would say. Given that types are only used in that struct, I prefer to have signatures available at one glance without having to go looking at the typedef.

Reply:
Interesting idea. But I don't think a typedef would make this part of the API cleaner.

Typical using code looks like this:

```
int my_custom_skip(void *f, dumb_off_t n)
{
    // custom implementation around f and n 
}
DUMBFILE_SYSTEM dfs;
dfs.skip = my_custom_skip;
```

Even if this had typedefs, the typedefs will never show in usercode, because it's not possible in C to define an implementation à la: `dumb_skip_func_t my_custom_skip { /* ... */ }`. You have to type the raw function signature.

The typedefs would merely make it slightly harder to chase to the definition?

Reply:
I think this would make it easier to see the actual callback names from all that mess. If the typedefs were direcly above the struct, they wouldn't be too hard to find. In any case, this would be purely cosmetic and I don't care enough to argue one way or another :)

Reply:
If readability of the struct members is at stake: How about a 1-line comment above each pointer? I.e., have this in `dumb.h`:

```
typedef struct DUMBFILE_SYSTEM {
    /* open: open a filename, return file as user-defined type */
    void *(*open)(const char *filename);

    /* skip: ignore the next n bytes in file f */
    int (*skip)(void *f, dumb_off_t n);
    
    /* ... */
}
```

And still point to DUMBFILE_SYSTEM.md for completeness.

Reply:
More documentation is always good 👍